include/pybind11/numpy.h

```diff
@@ -767,6 +769,9 @@ template <typename T, int ExtraFlags = array::forcecast> class array_t : public
     explicit array_t(ShapeContainer shape, const T *ptr = nullptr, handle base = handle())
         : array(std::move(shape), ptr, base) { }
 
+    explicit array_t(size_t count, const T *ptr = nullptr, handle base = handle())
```

Comment:
It's not exactly new ([here's the same constructor in 2.1.1](https://github.com/pybind/pybind11/blob/v2.1/include/pybind11/numpy.h#L513)), it just disappeared between #788 and #822.  I tried to handle it in #788 by having any_container accept single values, but that broke 2-element initializer_list construction (due to ambiguity; see #822), so #822 reintroduced the older singleton value constructors.

Yes, I think throwing an exception here for a negative value is warranted; `std::invalid_argument`?